xla/service/gpu/conv_algorithm_picker_test.cc

```diff
@@ -125,5 +140,105 @@ ENTRY main {
+  filter = f8e4m3fn[16,3,3,128] parameter(1)
+  input_scale = f32[] parameter(2)
+  input_scale_bcast = f32[1,6,6,128] broadcast(input_scale), dimensions={}
+  filter_scale = f32[] parameter(3)
+  filter_scale_bcast = f32[16,3,3,128] broadcast(filter_scale), dimensions={}
+  input_f32 = f32[1,6,6,128] convert(input)
+  input_unscaled = f32[1,6,6,128] multiply(input_f32, input_scale_bcast)
+  filter_f32 = f32[16,3,3,128] convert(filter)
+  filter_unscaled = f32[16,3,3,128] multiply(filter_f32, filter_scale_bcast)
+  conv_a = f32[1,6,6,16] convolution(input_unscaled, filter_unscaled), window={size=3x3 pad=1_1x1_1}, dim_labels=b01f_o01i->b01f, feature_group_count=1
+  z_scale = f32[] parameter(4)
+  z_scale_bcast = f32[1,6,6,16] broadcast(z_scale), dimensions={}
+  conv_a_scaled = f32[1,6,6,16] multiply(conv_a, z_scale_bcast)
+  c1 = f32[] constant(-448.)
+  c1_bcast = f32[1,6,6,16] broadcast(c1), dimensions={}
+  c2 = f32[] constant(448.)
+  c2_bcast = f32[1,6,6,16] broadcast(c2), dimensions={}
+  conv_a_clamped = f32[1,6,6,16] clamp(c1_bcast, conv_a_scaled, c2_bcast)
+  conv_a_clamped_f8 = f8e4m3fn[1,6,6,16] convert(conv_a_clamped)
+  abs_conv_a = f32[1,6,6,16] abs(conv_a)
+  c0 = f32[] constant(-inf)
+  amax = f32[] reduce(abs_conv_a, c0), dimensions={0,1,2,3}, to_apply=apply
+  ROOT conv_f8 = (f8e4m3fn[1,6,6,16], f32[]) tuple(conv_a_clamped_f8, amax)
+})";
+  TF_ASSERT_OK_AND_ASSIGN(auto m, ParseAndReturnVerifiedModule(kHlo));
+
+  se::Platform* platform = PlatformUtil::GetDefaultPlatform().value();
+  TF_ASSERT_OK_AND_ASSIGN(std::vector<se::StreamExecutor*> executors,
+                          PlatformUtil::GetStreamExecutors(platform));
+  ASSERT_GT(executors.size(), 0);
+  se::StreamExecutor* stream_exec = executors[0];
+
+  const se::GpuComputeCapability& cc = GetCudaComputeCapability();
+  bool changed = false;
+  TF_ASSERT_OK_AND_ASSIGN(changed, RunHloPass(GpuConvRewriter(cc), m.get()));
+  TF_ASSERT_OK_AND_ASSIGN(
+      changed, RunHloPass(CudnnFusedConvRewriter(GetCudaComputeCapability(),
+                                                 GetDnnVersion(), CUDA_VERSION),
+                          m.get()));
+
+  changed = false;
+  DebugOptions opts = DefaultDebugOptionsIgnoringFlags();
+
+  AutotuneConfig cfg{DeviceConfig{stream_exec, nullptr}, opts};
+  TF_ASSERT_OK_AND_ASSIGN(changed,
+                          RunHloPass(GpuConvAlgorithmPicker(cfg), m.get()));
+  ASSERT_TRUE(changed);
+
+  AutotuneResults results;
+  TF_ASSERT_OK(AutotunerUtil::SerializeAutotuneResults(&results));
+  ASSERT_EQ(results.results_size(), 1);
+  auto& result = *results.mutable_results(0)->mutable_result();
+  int64_t old_scratch_bytes = result.scratch_bytes();
+  int64_t new_scratch_bytes = old_scratch_bytes + 1;
+  result.set_scratch_bytes(new_scratch_bytes);
+
+  AutotunerUtil::ClearAutotuneResults();
+  TF_ASSERT_OK(AutotunerUtil::LoadAutotuneResults(results));
+
+  // Now send the same module through GpuConvAlgorithmPicker again.  The conv
```

Comment:
This test looks like it could benefit from being broken up into multiple ones.

Reply:
Do you have an idea how to split this? Currently, the test verifies that the convolution algorithm picker makes changes and that it correctly identifies the convolution and its scratch space when loading the results of a previous run, which requires running the algorithm picker and serializing the results. This is essentially the FP8 version of the existing SetAlgorithm unit test.